modules/dists/SparseBlockDist.chpl

```diff
@@ -327,6 +327,16 @@ class SparseBlockDom: BaseSparseDomImpl(?) {
       dsiAdd(i);
   }
 
+  proc setLocalSubdomain(locIndices, loc: locale = here) {
+    if loc != here then
+      halt("setLocalSubdomain() doesn't currently support remote updates");
```

Comment:
Since `loc` must be `here`, I am wondering what the point of having it at all as a formal is. It seems to me that if we support for setting the remote part of a domain here, calling it a local subdomain seems confusing, since it's not local, it's remote. (I can also see "it's local to the remote locale" but isn't everything local to somewhere?)

Reply:
My thinking was that we can say `MyDistArr.localSubdomain(michaelsLocale)` today, so I was trying to mimic that here (i.e., I can not only query a remote subdomain, but I can set one).  And then the part of this story I'm not proud of:  Being short on time, I was being too lazy to add tests for it (though I suspect they'd just work), so took this approach instead, opting for the interface that felt most correct and symmetric with our queries to me even though it's not wired up yet.

I could either add tests for this to check that it works, or remove the argument for now since it's not useful and could be added later without big impact.

Reply:
`MyDistArr.localSubdomain(michaelsLocale)` working was not obvious to me. Fine with me to proceed with what you have here but a Future Work section of the PR message might be a good place to mention that these don't work right now.

Reply:
I told a half-truth here.  The `.localSubdomain()` on domains and arrays in general takes a locale argument; however it seems that the dsi implementation of the routine on block sparse domains/arrays isn't actually hooked up to work.  I hadn't noticed this until I started looking into adding a test that would check how far off my implementation was from working.  I ended up putting this in the future work section instead.